tamr_client/project.py

```diff
@@ -55,5 +62,56 @@ def _from_json(url: URL, data: JsonDict) -> Project:
         return mastering_project._from_json(url, data)
     elif proj_type == "CATEGORIZATION":
         return categorization_project._from_json(url, data)
+    elif proj_type == "SCHEMA_MAPPING_RECOMMENDATIONS":
+        return schema_mapping_project._from_json(url, data)
     else:
         raise ValueError(f"Unrecognized project type '{proj_type}' in {repr(data)}")
+
+
+def create(
+    session: Session,
+    instance: Instance,
+    name: str,
+    proj_type: str,
+    description: str = None,
+    external_id: str = None,
+    unified_dataset_name: str = None,
+) -> Project:
+    """Create a project in Tamr.
+    Args:
+        instance: Tamr instance
+        name: Project name
+        proj_type: Project type
+        unified_dataset_name: Name of the unified dataset
+        description: (Optional) Project description
+        external_id: (Optional) External ID of the project
+
+    Returns:
+        Project created in Tamr
+
+    Raises:
+        Conflict: If a project with these specifications already exists.
+    """
+    if not unified_dataset_name:
+        unified_dataset_name = name + "_unified_dataset"
+    data = [
+        {
+            "name": name,
+            "type": proj_type,
+            "unifiedDatasetName": unified_dataset_name,
+            "description": description,
+            "externalId": external_id,
+        }
+    ]
```

Comment:
I think the [type error](https://github.com/Datatamer/tamr-client/pull/435/checks?check_run_id=985999724) you are seeing is because `data` is a `List`, but later on you try to do `data["relativeId"]`, which is trying to access an element of a list via string key. Did you mean to exclude the `[]` when defining `data`?

```suggestion
    data = {
        "name": name,
        "type": proj_type,
        "unifiedDatasetName": unified_dataset_name,
        "description": description,
        "externalId": external_id,
    }
```

Reply:
updated